src/services/widgets/widgets.svc.coffee

```diff
@@ -98,32 +98,26 @@ angular
 
     # TODO: move logic in ImpacDashboardsSvc
     # Sets setting for all widgets with same name
-    @updateAllSameWidgets = (dashboard, settings) ->
-      # Hack for tests correct working
-      _self.load().then ->
-        # Find name of parameter
-        paramName = _.keys(settings.toMetadata())[0];
-
-        # Write new settings metadata to current dashboard
-        ImpacDashboardsSvc.update(dashboard.id, settings.toMetadata()).then (updatedDashboard)->
-
-          # Update new setting on current dashboard
-          dashboard[paramName] = updatedDashboard[paramName]
-          _.each dashboard.widgets, (wgt) ->
-            # Retrieve the name of parameters attached to the widget
-            # TODO: export to a helper function in WidgetsSvc
-            wgtSettingsKeys = _.uniq( _.map( wgt.settings, (st) ->
-              _.keys(st.toMetadata())[0]
-            ))
-
-            # The widget's metadata are updated only if the correct setting is attached to the widget
-            if _.includes(wgtSettingsKeys, paramName)
-              wgt.metadata[paramName] = _.cloneDeep settings.toMetadata()[paramName]
-              wgt.isLoading = true
-              _self.update(wgt, { metadata: wgt.metadata }).then(
-                (updatedWidget) ->
-                  _self.show(updatedWidget).finally( -> updatedWidget.isLoading = false )
-              )
+    @updateAllSimilarWidgets = (dashboard, setting) ->
+      # Find setting key
+      settingKey = _.keys(setting.toMetadata())[0]
+
+      angular.extend dashboard.metadata, setting.toMetadata()
```

Comment:
you can use angular.extend, which is like a merge, but which will replace the setting instead of deep merging it